crates/autopilot/src/on_settlement_event.rs

```diff
@@ -302,3 +202,230 @@ impl Inner {
         }
     }
 }
+
+#[async_trait]
+impl OnSettlementEventAction for Updater {
+    async fn apply(&self) -> Result<bool> {
+        let mut ex = self
+            .db
+            .pool
+            .begin()
+            .await
+            .context("acquire DB connection")?;
+        let event = match database::settlements::get_settlement_without_auction(&mut ex)
```

Comment:
The `OnSettlementEventUpdater` is a weird mix between something that usually runs on every new settlement but should also have the ability to quickly go through a backlog of events (e.g. when we re-index a bunch of old events).
The new way would not allow for quickly working through a backlog because we always wait for the next settlements to happen.